src/Games/NexusMods.Games.RedEngine/Cyberpunk2077/SortOrder/RedMod/RedModSortableItemProvider.cs

```diff
@@ -0,0 +1,388 @@
+using System.Collections.ObjectModel;
+using DynamicData;
+using DynamicData.Kernel;
+using NexusMods.Abstractions.Games;
+using NexusMods.Abstractions.Loadouts;
+using NexusMods.Abstractions.Loadouts.Extensions;
+using NexusMods.Extensions.BCL;
+using NexusMods.Games.RedEngine.Cyberpunk2077.Models;
+using NexusMods.MnemonicDB.Abstractions;
+using NexusMods.MnemonicDB.Abstractions.TxFunctions;
+using NexusMods.Paths;
+using R3;
+
+namespace NexusMods.Games.RedEngine.Cyberpunk2077.SortOrder;
+
+public class RedModSortableItemProvider : ILoadoutSortableItemProvider, IDisposable
+{
+    private readonly IConnection _connection;
+
+    private readonly SourceCache<RedModSortableItem, string> _orderCache = new(item => item.RedModFolderName);
+    private readonly ReadOnlyObservableCollection<ISortableItem> _readOnlyOrderList;
+    private readonly ReadOnlyObservableCollection<RedModLoadoutGroup.ReadOnly> _redModsGroups;
+    private readonly SortOrderId _sortOrderId;
+    private readonly SemaphoreSlim _semaphore = new(1, 1);
+    private readonly CompositeDisposable _disposables = new();
+
+    public ReadOnlyObservableCollection<ISortableItem> SortableItems => _readOnlyOrderList;
+
+
+    public LoadoutId LoadoutId { get; }
+    public ISortableItemProviderFactory ParentFactory { get; }
+
+    public static async Task<RedModSortableItemProvider> CreateAsync(
+        IConnection connection,
+        LoadoutId loadoutId,
+        ISortableItemProviderFactory parentFactory)
+    {
+        var sortOrderModel = await GetOrAddSortOrderModel(connection, loadoutId, parentFactory);
+        return new RedModSortableItemProvider(connection,
+            loadoutId,
+            sortOrderModel,
+            parentFactory
+        );
+    }
+
+    private RedModSortableItemProvider(
+        IConnection connection,
+        LoadoutId loadoutId,
+        RedModSortOrder.ReadOnly sortOrderModel,
+        ISortableItemProviderFactory parentFactory)
+    {
+        _connection = connection;
+        LoadoutId = loadoutId;
+        ParentFactory = parentFactory;
+        _sortOrderId = sortOrderModel.AsSortOrder().SortOrderId;
+
+        // load the previously saved order
+        var order = RetrieveSortOrder();
```

Comment:
Might be a good idea to rename this to `RetrieveSortableItems()`.
`RetrieveSortOrder` may be confusing given the existing presence of `sortOrderModel`, which itself is the 'sort order', so to speak.

Reply:
I changed `PersistOrder()` and `RetrieveSortOrder()` to `PersistSortableEntries()` and `RetrieveSortableEntries()`.
What do you think?

Reply:
Yeah, that works well for me 👍